server/resources/resource_management_test.go

```diff
@@ -1292,6 +1305,27 @@ var _ = Describe("Resource manager", func() {
 					Expect(ok).To(BeTrue())
 					Expect(theResource).To(HaveKeyWithValue("test_string", "Ia, ia, HJPEV fhtang!"))
 				})
+
+				It("Should only modify updated fields", func() {
```

Comment:
name is misleading: you never actually check if the updated field is modified or not. please add an assert on `vlan_id`